shim.py

```diff
@@ -225,19 +280,39 @@ def removeObject(obj, existingRecords):
     logger.error("Failed to remove from list: %s" %(str(result)))
 
 def handleList(newGlobalList, existingRecords):
+  now = int(time.time())
   toAdd = set([x for x in newGlobalList if x not in globalList])
-  toRemove = set([x for x in globalList if x not in newGlobalList])
+
+  # Candidates for removal are owned but not currently labeled
+  removalCandidates = set([x for x in globalList if x not in newGlobalList])
+  toRemove = set()
+  for candidate in removalCandidates:
+    last_seen = globalLastSeen.get(candidate)
+    if last_seen is None:
+      # If unknown, initialize now to avoid immediate removal
+      globalLastSeen[candidate] = now
+      last_seen = now
+    age = now - last_seen
+    if age >= reapSeconds:
+      toRemove.add(candidate)
+    else:
+      remaining = reapSeconds - age
+      logger.info("Deferring removal for %s, reaping in ~%ss" %(str(candidate), remaining))
+
   toSync = set([x for x in globalList if ((x not in existingRecords["dns"]) and (x not in existingRecords["cname"]))])
```

Comment:
```suggestion

toSync = set([x for x in globalList if ((x not in existingRecords["dns"]) and (x not in existingRecords["cname"]))]) - toAdd- toRemove
```

Reply:
also needs line 319 to change  

`for sync in (toSync):`

Just avoids a false lo of syncs that never occur

Reply:
done, thanks